### DISCUSSION TITLE
版本管理和发版节奏

### DISCUSSION THREAD

**Discussion by teacher-zhou** [AchievedOwner/TDesignBlazor#134] [问答] [answered]
## 发版定义
1. 向 master 进行一次 tag
2. 打包并推送到 `nuget.org`
3. 产生 release note

### 方案 1
* 每一次 pull request 合并后，进行发版
* 规则：可以根据规则设定版本生成，比如功能是次版本号自增，bugfix 则是 小版本号自增等
* 触发：自动
* 优势：会有很多的小版本进行每天发布，感觉更适合 preview 版本
* 劣势：可能发布有点太频繁

### 方案 2（默认 ☀️ ）
* 当人工对 master 进行标记 tag 后进行发版
* 规则：人工编写 tag
* 触发：手动
* 优势：择机发布，完全看心情，但是简单
* 劣势：需要人工操作，版本记录的内容会非常长

### 方案 3
* 在 PR 时根据 label 的定义进行发版
* 规则：自动生成版本号
* 触发：人工触发？
* 优势：定义了指定 label 的功能会发布
* 劣势：需要更多的时间进行管理，自动化脚本编写难度提升

还有什么更好的方案希望可以交流

**Comment by elvismile** [comment]
pull request一次 就自动发版一次 太频繁了，显得版本号更新太快，对用户也不友好
感觉tag一次发布一次可以，但是最好能定期打tag，有个相对固定的周期

**Reply by teacher-zhou** [reply]
> pull request一次 就自动发版一次 太频繁了，显得版本号更新太快，对用户也不友好 感觉tag一次发布一次可以，但是最好能定期打tag，有个相对固定的周期

这个周期怎么定？固定时间，比如每几周一次？如果时这样的话，是不是可以搞成自动发布

**Reply by wuxinheng** [reply]
两周一次（开发进度慢），或者一周（开发进度快），能确认没什么问题自动发布，不能就手动啊

**Comment by teacher-zhou** [comment, marked as answer]
所以我们使用2周一个版本发布周期？

**Reply by teacher-zhou** [reply]
采取 方案2 的形式，并且周期定位，2周发布一次版本。
